index.d.ts

```diff
@@ -59,6 +59,6 @@ export declare const koaV1Middleware: (
 
 export declare const hapiPlugin: IHapiPlugin<IOptions>
 
-export declare const runWithId: (fn: Function, id?: any) => any
+export declare function runWithId<T>(fn: () => T, id?: string): T
```

Comment:
`id` should have `any` type. That's because Fastify uses numbers forrequest ids, or any other type if a custom id gen factory is provided.

Off-topic. It probably also makes sense to change return type of `id` function to `any`.

Reply:
Should be `unknown` at least

Reply:
@SimenB based on the description of [this PR](https://github.com/Microsoft/TypeScript/pull/24439), anything is assignable to `unknown`, so it `any` is the same thing as `uknown` when it's used as the argument. The difference reveals itself when it's used as the return type.

Reply:
Does it make sense to use `unknown` for function arguments? It seems that it's OK to use it for `id()` return type, but I'm not sure about the `id` argument of `runWithId()`.

Reply:
Anything is assignable to `unknown` yes, but it's more accurate than `any` - even if the type is only in the declaration and no code actually consumes the `unknown` type

Reply:
Also, if `id` function should change from returning a string it should definitely be `unknown`, and IMO the types should be internally consistent event though `any` and `unknown` as arguments are interchangable

Reply:
In this case, I'm fine with using `unknown` type.

Reply:
Could you update the PR?

Reply:
@puzpuzpuz done!